Form layout—Documentation (HDS-4886) (#2947)
hashicorp/design-system: website/docs/components/form/layout/index.md

---

> title: Layout

Reviewer: I kept it Layout since the other pages don't use Form in the title — like "Form Primitives" but if you think we should have Form in the title we can add it

> title: Layout

Reviewer 2: TBH, I think Primitives should be "Form Primitives." In looking at all the other subcategories outside of Form, they all repeat the name, e.g., "Copy / Copy Button", "Table / Advanced Table", "Stepper / Stepper Nav", "Link / Inline Link" 

I think the other form pages are fine because their component names stand on their own and would work out of context, but for the generic ones (Primitives, Layout), it's hard to know what they're about, particularly on the Overview page.

description: The Form layout components are used as structural components to organize and space the structure and spacing between elements in forms.

> caption: The Form layout components are used as structural components to organize and space the structure and spacing between elements in forms.

Author: ```suggestion
caption: Form layout components create consistent structure and spacing for form elements.
```

Author: Note: Captions should be as brief as possible. Descriptions can be a bit longer though.

links:
  figma: >-
    https://www.figma.com/design/iweq3r2Pi8xiJfD9e6lOhF/branch/azcdumnXvCyGfSVfEVZhGH/HDS-Components-v2.0?node-id=83107-2070&p=f&t=CeumEO6vBV6h0IuD-0
  github: >-
    https://github.com/hashicorp/design-system/tree/main/packages/components/src/components/hds/form
related:
  - patterns/form-patterns
previewImage: assets/illustrations/components/form/layout.jpg
navigation:
  order: 98

>   keywords: ['add', 'alternate', 'keywords', 'here']

Author: @LilithJames-HDS I added some keywords. Could you please check and see if there are any more you think would be relevant? (Or if you think I included any that you don't think are correct?)

>   keywords: ['add', 'alternate', 'keywords', 'here']

Reviewer: These look good to me, I can't think of others to add here. Thank you!

---
title: Form Layout
description: Form layout components are used to create consistent structure and spacing for form elements.
caption: Components used to create consistent structure and spacing for form elements.
links:
  figma: >-
    https://www.figma.com/design/iweq3r2Pi8xiJfD9e6lOhF/branch/azcdumnXvCyGfSVfEVZhGH/HDS-Components-v2.0?node-id=83107-2070&p=f&t=CeumEO6vBV6h0IuD-0
  github: >-
    https://github.com/hashicorp/design-system/tree/main/packages/components/src/components/hds/form
related:
  - patterns/form-patterns
previewImage: assets/illustrations/components/form/layout.jpg
navigation:

>   label: Layout

Reviewer 2: [question] Would changing the label to "Form Layout" also change the name on the card in the Overview page? Out of context, this is a bit odd, but I'm not sure what controls this. 

<img width="1240" alt="Screenshot 2025-06-24 at 4 13 36 PM" src="https://github.com/user-attachments/assets/b4f14610-5382-4263-ac5e-5714ad263bee" />

Reviewer 3: Yes, in this case is similar to "Primitives" next to it. But it's a good point, we could consider to use the other title, for the cards (maybe @KristinLBradley can you create a ticket for this, to do as follow-up when we have time?)

Reviewer 2: @didoo Do you mean have the cards and sidebar nav item use the "title" instead of the navigation "label" or am I misunderstanding?

Reviewer 3: No, to have the cards to use the `title` instead of the `navigation.label`. If this doesn't work because we have other exceptions, we could split the `previewImage` and have `preview.image` and `preview.label`.

Author: Resolving to clean up
